channel-router/src/services/index.ts

```diff
@@ -0,0 +1,83 @@
+import { Context } from 'koa';
+import { forkJoin } from 'rxjs';
+import { configuration } from '../config';
+import { IChannelMap, ITransaction } from '../interfaces';
+import http from 'http';
+import { LoggerService } from '../helpers';
+
+// service layer: business logic
+export default async function Execute(context: Context): Promise<Context> {
+  const channelMap: IChannelMap[] = Object.assign(
+    new Array<IChannelMap>(),
+    JSON.parse(configuration.channelMap),
+  );
+  let transaction = {} as ITransaction;
+  transaction = Object.assign(transaction, context.request.body);
+
+  let channelCounter = 0;
+  await forkJoin(
+    channelMap.map((channel) => {
+      channelCounter++;
+      try {
+        return sendRule(channel.endpoint, transaction);
+      } catch (err) {
+        LoggerService.error(err);
+        return err;
+      }
+    }),
+  ).toPromise();
```

Comment:
Ideally, we should avoid importing the rxjs library, so we could do something like this:
```
const arrayOfPromises = channelMap.map(channel => new Promise(sendRule(channel.endpoint, transaction));
await Promise.all(arrayOfPromises)
```

This way we don't need to import a library to map thru promises.

Reply:
I prefer built-in functions as much as possible. So, it would be very good if we can achieve them without 3rd party libraries.